fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTray.kt

```diff
@@ -97,42 +128,57 @@ fun TabsTray(
             ) { position ->
                 when (Page.positionToPage(position)) {
                     Page.NormalTabs -> {
-                        if (displayTabsInGrid) {
-                            TabGrid(
-                                tabs = normalTabs,
-                                onTabClose = onTabClose,
-                                onTabMediaClick = onTabMediaClick,
-                                onTabClick = handleTabClick,
-                                onTabLongClick = onTabLongClick,
-                            )
+                        val showInactiveTabsDialog = showInactiveTabsAutoCloseDialog(inactiveTabs.size)
+                        var showAutoCloseDialog by remember { mutableStateOf(showInactiveTabsDialog) }
+
+                        val optionalInactiveTabsHeader: (@Composable () -> Unit)?
+                        if (inactiveTabs.isNotEmpty()) {
+                            optionalInactiveTabsHeader = {
+                                InactiveTabsList(
+                                    inactiveTabs = inactiveTabs,
+                                    expanded = inactiveTabsExpanded,
+                                    showAutoCloseDialog = showAutoCloseDialog,
+                                    onHeaderClick = onInactiveTabsHeaderClick,
+                                    onDeleteAllButtonClick = onDeleteAllInactiveTabsClick,
+                                    onAutoCloseDismissClick = {
+                                        onInactiveTabAutoCloseDialogCloseButtonClick()
+                                        showAutoCloseDialog = !showAutoCloseDialog
+                                    },
+                                    onEnableAutoCloseClick = {
+                                        onEnableInactiveTabAutoCloseClick()
+                                        showAutoCloseDialog = !showAutoCloseDialog
+                                    },
+                                    onTabClick = onInactiveTabClick,
+                                    onTabCloseClick = onInactiveTabClose,
+                                )
+                            }
+
+                            if (showInactiveTabsDialog) {
+                                onInactiveTabsAutoCloseDialogShown()
+                            }
                         } else {
-                            TabList(
-                                tabs = normalTabs,
-                                onTabClose = onTabClose,
-                                onTabMediaClick = onTabMediaClick,
-                                onTabClick = handleTabClick,
-                                onTabLongClick = onTabLongClick,
-                            )
+                            optionalInactiveTabsHeader = null
```

Comment:
nit: Switching these cases so that the short null case is first might be it a little easier to read